CMakeSettings.json

```diff
@@ -1,365 +1,468 @@
+        }
+      ],
+      "inheritEnvironments": [ "msvc_x86_x64" ]
+    },
+    {
+      "name": "x64-Release",
+      "generator": "Visual Studio 16 2019 Win64",
+      "configurationType": "Release",
+      "buildRoot": "${projectDir}\\..\\hlsl.bin.${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "-m -v:minimal",
+      "variables": [
+        {
+          "name": "DXC_BUILD_ARCH",
+          "value": "x64",
+          "type": "STRING"
+        },
+        {
+          "name": "CLANG_ENABLE_ARCMT",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "CLANG_ENABLE_STATIC_ANALYZER",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "CLANG_INCLUDE_TESTS",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_INCLUDE_TESTS",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "HLSL_INCLUDE_TESTS",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_TARGETS_TO_BUILD",
+          "value": "None",
+          "type": "STRING"
+        },
+        {
+          "name": "LLVM_INCLUDE_DOCS",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_INCLUDE_EXAMPLES",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "LIBCLANG_BUILD_STATIC",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_OPTIMIZED_TABLEGEN",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_REQUIRES_EH",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_APPEND_VC_REV",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_ENABLE_RTTI",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_ENABLE_EH",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_DEFAULT_TARGET_TRIPLE",
+          "value": "dxil-ms-dx",
+          "type": "STRING"
+        },
+        {
+          "name": "CLANG_BUILD_EXAMPLES",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "LLVM_REQUIRES_RTTI",
+          "value": "True",
+          "type": "BOOL"
+        },
+        {
+          "name": "CLANG_CL",
+          "value": "False",
+          "type": "BOOL"
+        },
+        {
+          "name": "CMAKE_SYSTEM_VERSION",
+          "value": "10.0.14393.0",
+          "type": "STRING"
+        },
+        {
+          "name": "ENABLE_SPIRV_CODEGEN",
+          "value": "True",
+          "type": "BOOL"
         }
-    ]
-}
\ No newline at end of file
+      ],
+      "inheritEnvironments": [ "msvc_x64_x64" ]
+    }
+  ]
+}
```

Comment:
> I'm really uncomfortable with the json change. Why was it included?

Because I have 2019 installed and cannot permit myself to install 2017 side-by-side.

I asked [before](https://github.com/microsoft/DirectXShaderCompiler/issues/2680#issuecomment-635499459) whether this change was okay:

> , as well as the necessary updates to CMakeSettings.json to get it to work properly under VS 2019. Let me know if these you want these upstreamed too.

Without explicit answer I opted to leave this in rather than make a separate PR straight away. That can - _obviously_ - easily be done at any point.

As for compatibility, that was mentioned in the PR description:

> In addition CMakeSettings.json is updated to support VS2019, though I am unable to test backwards compatibility with VS2017 and below.

Would you mind validating this? I'd be _very_ disappointed if Microsoft/Visual Studio break backwards compatibility this way, with no easy way out (such as a simple `version` field).
That said the documentation for VS 2017 and VS 2019 on variable layout is identical:
https://docs.microsoft.com/en-us/cpp/build/cmakesettings-reference?view=vs-2017
https://docs.microsoft.com/en-us/cpp/build/cmakesettings-reference?view=vs-2019
This leaves me rather surprised; how's the current config working in 2017?